Board.java

```diff
@@ -1,22 +1,20 @@
 import java.util.ArrayList;
 
 public class Board {
-    private String state;
-    private int dimension;
-
-    private final static int[][] win = {{0, 1, 2}, {3, 4, 5}, {6, 7, 8}, {0, 3, 6}, {1, 4, 7}, {6, 7, 8}, {0, 4, 8}, {2, 4, 6}, {2, 4, 8}};
+    private final String state;
+    public final int dimension;
 
     public Board(String state) {
         this.state = state;
-        dimension = state.length();
+        dimension = state.length() / (int) Math.sqrt(state.length());
     }
 
     public String getState() {
         return this.state;
     }
 
      public Boolean availablePosition(int position) {
-         return position >= 0 && position <= dimension && state.charAt(position) == '-';
+         return position >= 0 && position <= dimension * dimension && state.charAt(position) == '-';
```

Comment:
you could pull out _position >= 0 && position  <= dimension * dimension) into a method to make this read better, I'll let you think of a relevant name.